tools/mpremote/mpremote/transport_serial.py

```diff
@@ -746,15 +751,18 @@ def __mount():
 """
 
 # Apply basic compression on hook code.
-for key, value in fs_hook_cmds.items():
-    fs_hook_code = re.sub(key, str(value), fs_hook_code)
 fs_hook_code = re.sub(" *#.*$", "", fs_hook_code, flags=re.MULTILINE)
 fs_hook_code = re.sub("\n\n+", "\n", fs_hook_code)
 fs_hook_code = re.sub("    ", " ", fs_hook_code)
 fs_hook_code = re.sub("rd_", "r", fs_hook_code)
 fs_hook_code = re.sub("wr_", "w", fs_hook_code)
 fs_hook_code = re.sub("buf4", "b4", fs_hook_code)
 
+# Replace hook command names with the integer command index.
+# Handle keys in reverse sort order to to avoid eg. READ replacing half of READLINE
+for key, value in reversed(sorted(fs_hook_cmds.items())):
+    fs_hook_code = re.sub(key, str(value), fs_hook_code)
```

Comment:
Maybe it's better to make a more sophisticated regular expression that matches stand and end of word, eg `r"\W" + key + r"\W"`.

Reply:
Yeah I had thought that briefly... it would be more explicit to fix the regex